sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -39,29 +40,362 @@ class ConfigurationSetting(Model):
+            tags=self.tags,
+            locked=self.read_only,
+            etag=self.etag,
+        )
+
+
+class FeatureFlagConfigurationSetting(
+    ConfigurationSetting
+):  # pylint: disable=too-many-instance-attributes
+    """A feature flag configuration value.
+    Variables are only populated by the server, and will be ignored when
+    sending a request.
+    :ivar etag: Entity tag (etag) of the object
+    :vartype etag: str
+    :ivar key:
+    :vartype key: str
+    :ivar value: The value of the configuration setting
+    :vartype value: str
+    :ivar enabled:
+    :vartype enabled: bool
+    :param filters:
+    :type filters: list[dict[str, Any]]
+    :param label:
+    :type label: str
+    :param content_type:
+    :type content_type: str
+    :ivar last_modified:
+    :vartype last_modified: datetime
+    :ivar read_only:
+    :vartype read_only: bool
+    :param tags:
+    :type tags: dict[str, str]
+    """
+
+    _attribute_map = {
+        "etag": {"key": "etag", "type": "str"},
+        "key": {"key": "key", "type": "str"},
+        "label": {"key": "label", "type": "str"},
+        "content_type": {"key": "content_type", "type": "str"},
+        "value": {"key": "value", "type": "str"},
+        "last_modified": {"key": "last_modified", "type": "iso-8601"},
+        "read_only": {"key": "read_only", "type": "bool"},
+        "tags": {"key": "tags", "type": "{str}"},
+    }
+    key_prefix = ".appconfig.featureflag/"
+    _feature_flag_content_type = (
+        "application/vnd.microsoft.appconfig.ff+json;charset=utf-8"
+    )
+    kind = "FeatureFlag"
+
+    def __init__(self, key, enabled, filters=None, **kwargs):
+        # type: (str, bool, Optional[List[Dict[str, Any]]]) -> None
+        self._enabled = enabled
+        super(FeatureFlagConfigurationSetting, self).__init__(**kwargs)
+        self.key = key
+        self.label = kwargs.get("label", None)
+        self.content_type = kwargs.get("content_type", self._feature_flag_content_type)
+        self.last_modified = kwargs.get("last_modified", None)
+        self.tags = kwargs.get("tags", {})
+        self.read_only = kwargs.get("read_only", None)
+        self.etag = kwargs.get("etag", None)
+        self.description = kwargs.get("description", None)
+        self.display_name = kwargs.get("display_name", None)
+        self._filters = filters or []
+        self._value = kwargs.get("value", {})
+
+    @property
+    def enabled(self):
+        # type: () -> bool
+        if self._value is None:
+            self._value = {"enabled": self._enabled}
+        elif not isinstance(self._value, dict):
+            raise AttributeError("value parameter is expected to be a dictionary")
+        else:
+            self._value["enabled"] = self._enabled
+        return self._enabled
+
+    @enabled.setter
+    def enabled(self, value):
+        # type: (bool) -> bool
+        if self._value is None:
+            self._value = {"enabled": value}
```

Comment:
If self._value is None, it is not a FeatureFlagConfigurationSetting, right? :)

Reply:
FeatureFlagConfigurationSetting is determined by the key_prefix and content_type, The value can be anything, there is no service side validation.

Reply:
Do you mean enabled property is Nullable?

Reply:
Yes it is nullable